### PR TITLE
DAOS-7904 net: Enforce client/server SRX match

### PR DESCRIPTION
**Pull request by mjmac**
If the server has set a value for FI_OFI_RXM_USE_SRX,
then the client must use the same value. If the server
has not set a value, then the client must not set one
either.

### CODE REVIEW

File: src/control/server/server_utils.go [hunk 1]
```diff
@@ -387,6 +388,68 @@ type netInterface interface {
...
+							return -1, srxMismatchErr
+						}
+						cliSrx = engSrx
+						continue
+					}
+					if cliSrx == engSrx {
+						continue
+					}
+					return -1, srxMismatchErr
+				}
```
Comment (kjacque) [inline src/control/server/server_utils.go:435]: A lot of nesting, a bit hard to follow. Here's a suggestion for breaking it into simpler non-nested statements.

```
if cliSrx == engSrx {
        continue
}
// no engine srx, or client srx is set but not equal
if engSrx == -1 || cliSrx != -1 || idx > 0 {
        return -1, srxMismatchErr
}
cliSrx = engSrx
```

File: src/control/server/server_utils.go [hunk 2]
```diff
@@ -387,6 +388,68 @@ type netInterface interface {
...
+			} else {
+				if cliSrx != -1 {
```
Comment (kjacque) [inline src/control/server/server_utils.go:437]: In a similar vein to above, could make this an `else if` to avoid unneeded nesting here.

**Review by kjacque — COMMENTED** [review]
Overall looks good, a couple suggestions for cleanup.

**Review by tanabarr — APPROVED** [review]
notwithstanding Kris' comments

File: src/control/server/server_utils.go (hunk 1 above)
Comment (mjmac) [reply src/control/server/server_utils.go:435]: Good call, thanks. I got it working but didn't take the time to make it understandable. This makes it easier to follow.

**Review by kjacque — DISMISSED** [review]

**Review by tanabarr — APPROVED** [review]

**Review by tanabarr — APPROVED** [review]

**Review by kjacque — APPROVED** [review]
